sweep/aggregator.go

```diff
@@ -0,0 +1,349 @@
+package sweep
+
+import (
+	"sort"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+const (
+	// DefaultFeeRateBucketSize is the default size of fee rate buckets
+	// we'll use when clustering inputs into buckets with similar fee rates
+	// within the SimpleAggregator.
+	//
+	// Given a minimum relay fee rate of 1 sat/vbyte, a multiplier of 10
+	// would result in the following fee rate buckets up to the maximum fee
+	// rate:
+	//
+	//   #1: min = 1 sat/vbyte, max = 10 sat/vbyte
+	//   #2: min = 11 sat/vbyte, max = 20 sat/vbyte...
+	DefaultFeeRateBucketSize = 10
+)
+
+// UtxoAggregator defines an interface that takes a list of inputs and
+// aggregate them into groups. Each group is used as the inputs to create a
+// sweeping transaction.
+type UtxoAggregator interface {
+	// ClusterInputs takes a list of inputs and groups them into clusters.
+	ClusterInputs(pendingInputs) []inputCluster
```

Comment:
public interface but operating on private types? don't think they're type aliases, so can't actually be called outside the pkg

Reply:
good point, will fix this in the final PR to avoid too many rebase conflicts